src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509CertificateCollection.cs

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Collections;
+using System.Diagnostics;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+using Interlocked = System.Threading.Interlocked;
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    public partial class X509CertificateCollection : ICollection, IEnumerable, IList
+    {
+        public X509CertificateCollection()
+        {
+            _list = new LowLevelListWithIList<Object>();
+        }
+
+        public X509CertificateCollection(X509Certificate[] value)
+            : this()
+        {
+            this.AddRange(value);
+        }
+
+        public X509CertificateCollection(X509CertificateCollection value)
+            : this()
+        {
+            this.AddRange(value);
+        }
+
+        public int Count
+        {
+            get { return _list.Count; }
+        }
+
+        bool ICollection.IsSynchronized
+        {
+            get { return false; }
+        }
+
+        Object ICollection.SyncRoot
+        {
+            get
+            {
+                return Interlocked.CompareExchange(ref _syncRoot, new Object(), null);
```

Comment:
This will allocate a new object on every access, even if _syncRoot is already non-null.